dosagelib/plugins/a.py

```diff
@@ -136,6 +136,45 @@ class ALessonIsLearned(_BasicScraper):
     help = 'Index format: nnn'
 
 
+class Alfie(WordPressScraper):
+    url = 'https://buttsmithy.com/'
+    stripUrl = url + 'archives/comic/%s'
+    firstStripUrl = stripUrl % 'p1'
+    adult = True
+
+    def namer(self, image_url, page_url):
+        def repl(m):
+            return "{0}".format(m.group(1).zfill(4))
+
+        if page_url == "https://buttsmithy.com/":
+            page = self.getPage(page_url)
+            links = page.xpath('//h2[@class="post-title"]/a/@href')
+            page_url = links[0]
+
+        name = sub('^p-?(\d+)', repl, page_url.split('/')[-1])
+
+        if name == "o-525":
+            name = "0525"
+        if name == "0369-2":
+            name = "0469"
+        if name == "1404":
+            name = "0378"
+        if name == "0338-2":
+            name = "0339"
+        if page_url == "https://buttsmithy.com/archives/comic/1186":
+            name = "0324"
+        if page_url == "https://buttsmithy.com/archives/comic/1169":
+            name = "0319"
+        if page_url == "https://buttsmithy.com/archives/comic/1132":
+            name = "0313"
+        if page_url == "https://buttsmithy.com/archives/comic/p-145":
+            name = "0145-2"
+        if page_url == "https://buttsmithy.com/archives/comic/p145":
```

Comment:
Maybe put all those special cases into a dict and loop over them instead of having big decision trees?

Reply:
Made it a dictionary and a loop.